modulus/models/graphcast/graph_cast_net.py

```diff
@@ -173,35 +170,108 @@ def __init__(
         self.mesh_edata = self.mesh_graph.edata["x"]
         self.mesh_ndata = self.mesh_graph.ndata["x"]
 
-        if use_cugraphops_encoder:
-            offsets, indices, edge_ids = self.g2m_graph.adj_tensors("csc")
-            n_in_nodes, n_out_nodes = (
+        if use_cugraphops_encoder or self.is_distributed:
+            print(self.g2m_graph)
+            try:
+                offsets, indices, edge_ids = self.g2m_graph.adj_tensors("csc")
+            except:
+                # fall back to older DGL routine
+                offsets, indices, edge_ids = self.g2m_graph.adj_sparse("csc")
+
+            n_src_nodes, n_dst_nodes = (
                 self.g2m_graph.num_src_nodes(),
                 self.g2m_graph.num_dst_nodes(),
             )
+            self.g2m_edata = self.g2m_edata[edge_ids]
             self.g2m_graph = CuGraphCSC(
-                offsets, indices, n_in_nodes, n_out_nodes, edge_ids
+                offsets,
+                indices,
+                n_src_nodes,
+                n_dst_nodes,
+                partition_size=partition_size,
+                partition_group_name=partition_group_name,
             )
-
-        if use_cugraphops_decoder:
-            offsets, indices, edge_ids = self.m2g_graph.adj_tensors("csc")
-            n_in_nodes, n_out_nodes = (
+            if self.is_distributed:
+                self.g2m_edata = self.g2m_graph.get_edge_features_in_partition(
+                    self.g2m_edata
+                )
+
+        if use_cugraphops_decoder or self.is_distributed:
+            try:
+                offsets, indices, edge_ids = self.m2g_graph.adj_tensors("csc")
+            except:
```

Comment:
Since this is a specific test in the try block, would be good to either catch only that specific exception so other exceptions still get raised or use something like `if hasattr(self.m2g_graph, "adj_tensors")`

Reply:
agreed, good suggestion